Syslog source IP test plan

JibinBao opened the pull request:
Add test plan for Syslog source IP test plan feature:

Syslog source IP HLD:https://github.com/nazariig/SONiC/blob/1f2f6713495d04b41d19e0f201bff72af60ef52f/doc/syslog/syslog-design.md

bingwang-ms committed:
Add FLAGS table

Signed-off-by: bingwang <bingwang@microsoft.com>

bingwang-ms committed:
Add clarification for init_cfg

Signed-off-by: bingwang <bingwang@microsoft.com>

bingwang-ms committed:
List existing flags

Signed-off-by: bingwang <bingwang@microsoft.com>

JibinBao committed:
Syslog source IP test plan

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
````diff
@@ -0,0 +1,165 @@
+### Test cases #3 -  Configure syslog server with VRF/Source: set/unset
+1. For vrf: Default, mgmt, Vrf-data, Configure syslog server with VRF/Source: set/unset like below:
+```
+config syslog add '2.2.2.2' --vrf "Default"
````
@JibinBao the keyword is `default` - please fix a typo

JibinBao replied:
will fix it. thanks

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
````diff
@@ -0,0 +1,165 @@
+```
+config syslog add '2.2.2.2' --vrf "Default"
+config syslog add '3.3.3.3' --vrf "mgmt"
+config syslog add '2222::2222' --vrf "Vrf-data"
````
@JibinBao this VRF suppose to exist in SONiC VRF table. Are you going to send syslog packets over the dataplane interface?

JibinBao replied:
Yes, we need to create a VRF(Vrf-data) for the test. 
I think after we add the above configuration for syslog server,  we don't need to send syslog packets with PTF tools, the dut will send the syslog message to the syslog server address automatically. Also, we can run the cmd of 'logger " syslog test message " ', it will send syslog message to the syslog server address.

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
```diff
@@ -0,0 +1,165 @@
+# show syslog
+SERVER      SOURCE      PORT    VRF
+----------  ----------  ------  --------
+2.2.2.2       N/A        N/A     Default
```
@JibinBao a typo: `Default` -> `default`

JibinBao replied:
will fix it

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
```diff
@@ -0,0 +1,165 @@
+
+
+### Test cases #3 -  Configure syslog server with VRF/Source: set/unset
+1. For vrf: Default, mgmt, Vrf-data, Configure syslog server with VRF/Source: set/unset like below:
```
@JibinBao a typo: `Default` -> `default`

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
````diff
@@ -0,0 +1,165 @@
+3.3.3.3       N/A        N/A     mgmt
+2222::2222    N/A        N/A     Vrf-data
+```
+3. Check the corresponding interface of rsyslog server can receive the syslog message with port 514
````
@JibinBao how it suppose to work for three different IPs for the same interface?

JibinBao replied:
I suppose different IPs on a different interface. 
Since the Syslog will be forwarded according to the route, can we make the route to the same interface if we use one interface?

nazariig replied:
@JibinBao yes. The route configuration should do the trick. But please double check, since on a t0 we have a default one (dataplane) plus ip rules.

JibinBao replied:
Actually, I haven't done any tests on the community test bed, and just do some simple tests on a canonical setup.
So I am not sure which topo will be suitable for these tests. 
Because we need to create vrf, I find the existing vrf tests only run t0 topo, so here I choose t0 topo.

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
````diff
@@ -0,0 +1,165 @@
+### Test cases #4 -  Configure syslog server with VRF/Source: set/set
+1. For vrf: Default, mgmt, Vrf-data, Configure syslog server with VRF/Source: set/set like below:
+```
+config syslog add '2.2.2.2' ---source "1.1.1.1" --vrf "Default"
````
@JibinBao a typo: `Default` -> `default`

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
```diff
@@ -0,0 +1,165 @@
+
+| **Document Name** | **Link** |
+|-------------------|----------|
+| Syslog Source IP HLD | [https://github.com/nazariig/SONiC/blob/1f2f6713495d04b41d19e0f201bff72af60ef52f/doc/syslog/syslog-design.md]|
```
@JibinBao has to be fixed once the HLD is merged to master

JibinBao replied:
yes

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
````diff
@@ -0,0 +1,165 @@
+3.3.3.3     5.5.5.5      N/A     mgmt
+2222::2222  1111::1111      N/A     Vrf-data
+```
+3. Check the corresponding interface of rsyslog server can receive the syslog message with correct port and source ip
````
@JibinBao how it suppose to work for three different IPs for the same interface?

JibinBao replied:
I suppose target different IPs on a different interface.
Since the Syslog will be forwarded according to the route, can we make the route to the same interface if we use one interface?

nazariig replied:
@JibinBao PSA

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
````diff
@@ -0,0 +1,165 @@
+### Test cases #5 -  Remove source IP config, check the relevant syslog config will be removed
+1. Configure syslog server with VRF/Source: set/set like below:
+```
+config syslog add '2.2.2.2' ---source "1.1.1.1" --vrf "Default"
````
@JibinBao a typo: `Default` -> `default`

JibinBao replied:
will fix it

JibinBao committed:
Fix typo error

bingwang-ms committed:
Update control-sonic-behaviors-with-sonic-flags.md

JibinBao committed:
Add corner case

JibinBao committed:
Remove 2 corner case, because it is not a requirement

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
```diff
@@ -147,40 +147,3 @@ config syslog del '2222::2222'
-4. Check the relevant syslog config should be removed
-
-
-### Test cases #6 -  Disable mgmt VRF or remove data VRF if this VRF exists in syslog config, there will an error prompt
```
@JibinBao since we do have a reference in YANG for this, i will add a relevant validation

JibinBao replied:
In the email thread, Assaf said that the case can be removed? 
Are you sure that we still keep it?

nazariig replied:
@JibinBao that is part of the design. Also, we have it in a YANG model. I think that removal was a mistake.

JibinBao replied:
Ok. I will add it back.

alexrallen committed:
Update references of /var/platform to /var/firmwareupdate to match implementation (#1000)

JibinBao committed:
Add one negative case

sg893052 committed:
SYSTEM READY (#977)

Support for System Ready ( Followup to #875)

Repo	                 PR title	                                          State
sonic-buildimage	 System                                              Ready	
sonic-utilities	         Show commands for System            Ready

JibinBao committed:
Change the steps for checking syslog msg

JibinBao committed:
Add reboot case for checking syslog still work

kannankvs committed:
Updated the build image link for commit diff (#1003)

As discussed, we have updated the build image with commit diff details for all the builds generated.

JibinBao committed:
Add 2 nagtive test case, remove single quote and double quote

bingwang-ms committed:
Add an example

Signed-off-by: bingwang <wang.bing@microsoft.com>

bingwang-ms committed:
Merge pull request #982 from bingwang-ms/add_flags_table

[HLD] Add `SYSTEM_DEFAULTS` table in `config_db` to control the behavior of SONiC

stepanblyschak committed:
[dump] Add ACL modules (#983)

Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>

shlomibitton committed:
Fast-reboot Flow Improvements HLD (#980)

Signed-off-by: Shlomi Bitton <shlomibi@nvidia.com>

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
```diff
@@ -0,0 +1,213 @@
+4. Check there is an error prompt such as: VRF exists in syslog config, it can not be disabled or removed.
+
+
+### Test cases #6 -  After Configure syslog server and save config, do cold/fast/warm reboot, check syslog config still work
```
@JibinBao please move to the end of the TCs list

JibinBao replied:
ok

JibinBao replied:
Done

nazariig commented on doc/syslog/Syslog_Source_IP_test_plan.md:
```diff
@@ -0,0 +1,213 @@
+5. Check Syslog config still work
+
+
+ ### Test cases #7 -  Configure syslog server with invalid source IP
```
@JibinBao suggest to verify the case when IP is not configured on any VRF member

JibinBao replied:
yes. in the step4(non-existing IP ), we have it.
4. Repeat step 1~3 with non-existing IP and multicast IP

JibinBao committed:
move Test cases #6 to the end of TCs list

JibinBao committed:
Change case #6

JibinBao committed:
Merge branch 'sonic-net:master' into Syslog_Source_IP
